api/cloud/oc_cloud_manager.c

```diff
@@ -287,14 +299,24 @@ cloud_register(void *data)
   return OC_EVENT_DONE;
 }
 
+static oc_cloud_error_t
+_login_handler_check_data_error(oc_client_response_t *data)
+{
+  if (data->code >= OC_STATUS_SERVICE_UNAVAILABLE) {
+    return CLOUD_ERROR_CONNECT;
+  }
+  if (data->code >= OC_STATUS_BAD_REQUEST) {
+    return CLOUD_ERROR_RESPONSE;
+  }
+  return CLOUD_OK;
+}
+
 static int
 _login_handler(oc_cloud_context_t *ctx, oc_client_response_t *data)
 {
-  if (data->code >= OC_STATUS_SERVICE_UNAVAILABLE) {
-    cloud_set_last_error(ctx, CLOUD_ERROR_CONNECT);
-    goto error;
-  } else if (data->code >= OC_STATUS_BAD_REQUEST) {
-    cloud_set_last_error(ctx, CLOUD_ERROR_RESPONSE);
+  oc_cloud_error_t err = _login_handler_check_data_error(data);
+  if (err != CLOUD_OK) {
+    cloud_set_last_error(ctx, err);
```

Comment:
pls remove the line - just `goto error` will exist - because the last error will be set in `error block`

Reply:
use in error section `cloud_set_cps_and_last_error(ctx, OC_CPS_FAILED, err);`